fs.go

```diff
@@ -39,21 +39,14 @@ type Filesystem interface {
 	MkdirAll(filename string, perm os.FileMode) error
 	Join(elem ...string) string
 	Dir(path string) Filesystem
-	Base() string
-}
-
-// Symlinker is a Filesystem with support for creating symlinks.
-type Symlinker interface {
```

Comment:
Since we are releasing go-git I prefer not having an independent interface. We can release the v3. 

Far as I know the only thrird party implementation is ours and we should support link on it.